diff --git a/gcc/config/haiku.h b/gcc/config/haiku.h
index 602eeca..e8664c5 100644
--- a/gcc/config/haiku.h
+++ b/gcc/config/haiku.h
@@ -97,43 +97,41 @@ Boston, MA 02111-1307, USA.  */
     { GCC_INCLUDE_DIR, "GCC", 0, 0, 0, 0 }, \
     { FIXED_INCLUDE_DIR, "GCC", 0, 0, 0, 0 }, \
     { TOOL_INCLUDE_DIR, "BINUTILS", 0, 1, 0, 0 }, \
-    { "/boot/system/non-packaged/develop/headers/" HYBRID_SECONDARY, 0, 0, 0, 1, 0 }, \
-    { "/boot/system/develop/headers/os", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/app", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/device", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/drivers", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/game", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/interface", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/kernel", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/locale", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/mail", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/media", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/midi", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/midi2", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/net", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/opengl", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/storage", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/support", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/translation", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/graphics", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/input_server", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/mail_daemon", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/registrar", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/screen_saver", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/tracker", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/Deskbar", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/NetPositive", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/Tracker", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/3rdparty", 0, 0, 0, 1, 0 }, \
-    { "/boot/system/develop/headers/bsd", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/glibc", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/gnu", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/posix", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/" HYBRID_SECONDARY, 0, 0, 0, 1, 0 }, \
+    { "/include/os", 0, 0, 1, 1, 0 }, \
+    { "/include/os/app", 0, 0, 1, 1, 0 }, \
+    { "/include/os/device", 0, 0, 1, 1, 0 }, \
+    { "/include/os/drivers", 0, 0, 1, 1, 0 }, \
+    { "/include/os/game", 0, 0, 1, 1, 0 }, \
+    { "/include/os/interface", 0, 0, 1, 1, 0 }, \
+    { "/include/os/kernel", 0, 0, 1, 1, 0 }, \
+    { "/include/os/locale", 0, 0, 1, 1, 0 }, \
+    { "/include/os/mail", 0, 0, 1, 1, 0 }, \
+    { "/include/os/media", 0, 0, 1, 1, 0 }, \
+    { "/include/os/midi", 0, 0, 1, 1, 0 }, \
+    { "/include/os/midi2", 0, 0, 1, 1, 0 }, \
+    { "/include/os/net", 0, 0, 1, 1, 0 }, \
+    { "/include/os/opengl", 0, 0, 1, 1, 0 }, \
+    { "/include/os/storage", 0, 0, 1, 1, 0 }, \
+    { "/include/os/support", 0, 0, 1, 1, 0 }, \
+    { "/include/os/translation", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/graphics", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/input_server", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/mail_daemon", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/registrar", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/screen_saver", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/tracker", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/Deskbar", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/NetPositive", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/Tracker", 0, 0, 1, 1, 0 }, \
+    { "/include/3rdparty", 0, 0, 0, 1, 0 }, \
+    { "/include/bsd", 0, 0, 1, 1, 0 }, \
+    { "/include/glibc", 0, 0, 1, 1, 0 }, \
+    { "/include/gnu", 0, 0, 1, 1, 0 }, \
+    { "/include/posix", 0, 0, 1, 1, 0 }, \
+    { "/include/" HYBRID_SECONDARY, 0, 0, 0, 1, 0 }, \
         /* Hybrid secondary folders for os kits not in base haiku package */\
-    { "/boot/system/develop/headers/" HYBRID_SECONDARY "/os", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/" HYBRID_SECONDARY "/os/opengl", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers", 0, 0, 0, 1, 0 }, \
+    { "/include/" HYBRID_SECONDARY "/os", 0, 0, 1, 1, 0 }, \
+    { "/include/" HYBRID_SECONDARY "/os/opengl", 0, 0, 1, 1, 0 }, \
     { 0, 0, 0, 0, 0, 0 } \
 }
 #else /* HYBRID_SECONDARY */
@@ -150,40 +148,37 @@ Boston, MA 02111-1307, USA.  */
     { GCC_INCLUDE_DIR, "GCC", 0, 0, 0, 0 }, \
     { FIXED_INCLUDE_DIR, "GCC", 0, 0, 0, 0 }, \
     { TOOL_INCLUDE_DIR, "BINUTILS", 0, 1, 0, 0 }, \
-    { "/boot/system/non-packaged/develop/headers", 0, 0, 0, 1, 0 }, \
-    { "/boot/system/develop/headers/os", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/app", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/device", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/drivers", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/game", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/interface", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/kernel", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/locale", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/mail", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/media", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/midi", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/midi2", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/net", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/opengl", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/storage", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/support", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/translation", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/graphics", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/input_server", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/mail_daemon", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/registrar", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/screen_saver", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/add-ons/tracker", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/Deskbar", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/NetPositive", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/os/be_apps/Tracker", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/3rdparty", 0, 0, 0, 1, 0 }, \
-    	/* TODO: To be removed when libtiff has been outsourced. */\
-    { "/boot/system/develop/headers/bsd", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/glibc", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/gnu", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers/posix", 0, 0, 1, 1, 0 }, \
-    { "/boot/system/develop/headers", 0, 0, 0, 1, 0 }, \
+    { "/include/os", 0, 0, 1, 1, 0 }, \
+    { "/include/os/app", 0, 0, 1, 1, 0 }, \
+    { "/include/os/device", 0, 0, 1, 1, 0 }, \
+    { "/include/os/drivers", 0, 0, 1, 1, 0 }, \
+    { "/include/os/game", 0, 0, 1, 1, 0 }, \
+    { "/include/os/interface", 0, 0, 1, 1, 0 }, \
+    { "/include/os/kernel", 0, 0, 1, 1, 0 }, \
+    { "/include/os/locale", 0, 0, 1, 1, 0 }, \
+    { "/include/os/mail", 0, 0, 1, 1, 0 }, \
+    { "/include/os/media", 0, 0, 1, 1, 0 }, \
+    { "/include/os/midi", 0, 0, 1, 1, 0 }, \
+    { "/include/os/midi2", 0, 0, 1, 1, 0 }, \
+    { "/include/os/net", 0, 0, 1, 1, 0 }, \
+    { "/include/os/opengl", 0, 0, 1, 1, 0 }, \
+    { "/include/os/storage", 0, 0, 1, 1, 0 }, \
+    { "/include/os/support", 0, 0, 1, 1, 0 }, \
+    { "/include/os/translation", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/graphics", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/input_server", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/mail_daemon", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/registrar", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/screen_saver", 0, 0, 1, 1, 0 }, \
+    { "/include/os/add-ons/tracker", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/Deskbar", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/NetPositive", 0, 0, 1, 1, 0 }, \
+    { "/include/os/be_apps/Tracker", 0, 0, 1, 1, 0 }, \
+    { "/include/3rdparty", 0, 0, 0, 1, 0 }, \
+    { "/include/bsd", 0, 0, 1, 1, 0 }, \
+    { "/include/glibc", 0, 0, 1, 1, 0 }, \
+    { "/include/gnu", 0, 0, 1, 1, 0 }, \
+    { "/include/posix", 0, 0, 1, 1, 0 }, \
     { 0, 0, 0, 0, 0, 0 } \
 }
 #endif /* HYBRID_SECONDARY */
@@ -202,12 +197,12 @@ Boston, MA 02111-1307, USA.  */
 #ifdef HYBRID_SECONDARY
 /* For a secondary compiler on a hybrid system, use alternative search paths.*/
 #define STANDARD_STARTFILE_PREFIX_1 \
-  "/boot/system/non-packaged/develop/lib/" HYBRID_SECONDARY "/"
+  "/lib/" HYBRID_SECONDARY "/"
 #define STANDARD_STARTFILE_PREFIX_2 \
-  "/boot/system/develop/lib/" HYBRID_SECONDARY "/"
+  "/lib/" HYBRID_SECONDARY "/"
 #else /* HYBRID_SECONDARY */
-#define STANDARD_STARTFILE_PREFIX_1   "/boot/system/non-packaged/develop/lib/"
-#define STANDARD_STARTFILE_PREFIX_2   "/boot/system/develop/lib/"
+#define STANDARD_STARTFILE_PREFIX_1   "/lib/"
+#define STANDARD_STARTFILE_PREFIX_2   "/lib/"
 #endif /* HYBRID_SECONDARY */
 
 /* Haiku doesn't have a separate math library.  */
diff --git a/gcc/config/t-haiku b/gcc/config/t-haiku
index ba1f7f5..90b27d2 100644
--- a/gcc/config/t-haiku
+++ b/gcc/config/t-haiku
@@ -1,6 +1,6 @@
 # There are system headers elsewhere, but these are the ones that
 # we are most likely to want to apply any fixes to.
-NATIVE_SYSTEM_HEADER_DIR = /boot/system/develop/headers
+NATIVE_SYSTEM_HEADER_DIR = /include
 LIMITS_H_TEST = [ -f $(SYSTEM_HEADER_DIR)/posix/limits.h ]
 
 # Haiku-specific attributes
